packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -254,7 +254,8 @@ class PigeonOptions {
           : null,
       swiftOut: map['swiftOut'] as String?,
       swiftOptions: map.containsKey('swiftOptions')
-          ? SwiftOptions.fromMap((map['swiftOptions'] as Map<String, Object>?)!)
+          ? SwiftOptions.fromList(
+              (map['swiftOptions'] as Map<String, Object>?)!)
```

Comment:
Optional nit: `fromList(map['swiftOptions']! as Map<String, Object>)`

(If you change this you should change the others in this call at the same time.)